### ISSUE TITLE
Invalid class parameters?

### ISSUE THREAD

**Issue by squarebracket** [echocat/puppet-graphite#259]
These seem to be new parameters in `manifests/init.pp`:

```
  $gr_base_dir                            = '/opt/graphite',
  $gr_storage_dir                         = "${gr_base_dir}/storage",
  $gr_local_data_dir                      = "${gr_storage_dir}/whisper",
  $gr_rrd_dir                             = "${gr_storage_dir}/rrd",
  $gr_whitelists_dir                      = "${gr_storage_dir}/lists",
  $gr_carbon_conf_dir                     = "${gr_base_dir}/conf",
  $gr_carbon_log_dir                      = "${gr_storage_dir}/log/carbon-cache",
  $gr_graphiteweb_log_dir                 = "${gr_storage_dir}/log",
  $gr_graphiteweb_conf_dir                = "${gr_base_dir}/conf",
  $gr_graphiteweb_webapp_dir              = "${gr_base_dir}/webapp",
  $gr_graphiteweb_storage_dir             = '/var/lib/graphite-web',
  $gr_graphiteweb_install_lib_dir         = "${gr_graphiteweb_webapp_dir}/graphite",
```

I'm using puppet 3.8, and it doesn't work because it won't interpolate parameters within the same class (i.e. all the parameters that use `${gr_*}` don't interpolate that variable).

Is this something that works in Puppet 4?

**Comment by dwerder** [comment]
As far as I know puppet should be able to interpolate paremeters since 2.7.
But this is a bad behavior and this parameters defaults should be set in params.pp

**Comment by squarebracket** [comment]
I already coded that up, you want me to make a PR?

**Comment by dwerder** [comment]
Sure

**Comment by dwerder** [comment]
Possible fix in #261

**Comment by dwerder** [comment]
I merged PR #262 into develop.
It fixes the parameter interpolation issue, without breaking it intention of changing all paths it `gr_base_dir` is set.

I also added a basic test for class config and fixed a lot of static dir path.
Can you try if it fixes the issue for you?

**Comment by dwerder** [comment]
New release is on the forge.
